contracts/modules/v2/StakingPool.sol

```diff
@@ -5,39 +5,26 @@ pragma solidity ^0.8.9;
 import "@openzeppelin/contracts-v4/token/ERC20/IERC20.sol";
 // TODO: consider using solmate ERC721 implementation
 import "@openzeppelin/contracts-v4/token/ERC721/ERC721.sol";
-import "@openzeppelin/contracts-v4/utils/math/SafeCast.sol";
 import "../../interfaces/IStakingPool.sol";
 import "../../interfaces/ICover.sol";
 import "../../interfaces/INXMToken.sol";
 import "../../utils/Math.sol";
+import "../../utils/SafeUintCast.sol";
+import "./StakingTypesLib.sol";
+import "./StakingTypesLib.sol";
 
 // total stake = active stake + expired stake
-// product stake = active stake * product weight
-// product stake = allocated product stake + free product stake
-// on cover buys we allocate the free product stake and it becomes allocated.
-// on expiration we deallocate the stake and it becomes free again
-
-// ╭───────╼ Active stake ╾────────╮
```

Comment:
where did the diagram go?

Reply:
not relevant to the new implementation anymore